dingo/gw/dataset/generate_dataset.py

```diff
@@ -194,12 +195,13 @@ def generate_dataset(settings: Dict, num_processes: int) -> WaveformDataset:
 
                 n_train = svd_settings["num_training_samples"]
                 n_test = svd_settings.get("num_validation_samples", 0)
-                parameters, polarizations = generate_parameters_and_polarizations(
-                    waveform_generator,
+
+                func = partial(
+                    generate_parameters_and_polarizations, 
+                    waveform_generator, 
```

Comment:
Can we pass all arguments of `generate_parameters_and_polarizations` explicitly
```suggestion
                    waveform_generator=waveform_generator, 
                    prior=prior,
```
to make it more robust to refactoring?

(Maybe we should ask @vincentberenz about the best practices related kwargs in general) :)

Reply:
Ah, that's an interesting point, I wasn't aware of this behavior.
Thanks for pointing it out!